### DISCUSSION TITLE
Transform code as little as possible

### DISCUSSION THREAD

**Discussion by allansson** [parcel-bundler/parcel#7225] [Q&A] [answered]
So we're making a web extension and I've wanted to try out parcel for a long while. And with the official web extension plugin, I figured that it's a no-brainer.

When developing a web extension you pretty much know your platform support, unlike with web development where browser support might vary wildly. Some browser vendors also require you to include a readable version of your code, so that they can do a review. So right now we're packaging up all our source code in a separate zip-file and include that when uploading new versions of the extension. 

It would be nice if we could configure parcel to keep the source code as close to the original as possible, e.g. strip type annotations from typescript, keep the file structure and use es modules for imports.

 Is this possible?

**Comment by mischnic** [comment, marked as answer]
Currently not really. Another reason to retain the file structure of the input is to still be able to leverage `sideEffects: false` when building libraries.

Related: https://github.com/parcel-bundler/parcel/issues/6092

**Reply by allansson** [reply]
That's too bad. At least it's not a deal breaker in my use-case.
